app/src/main/java/ar/com/wolox/android/example/ui/auth/login/AuthPresenter.kt

```diff
@@ -1,9 +1,25 @@
 package ar.com.wolox.android.example.ui.auth.login
 
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.core.util.ToastFactory
 import javax.inject.Inject
 
-class AuthPresenter @Inject constructor() : BasePresenter<AuthView>() {
+class AuthPresenter @Inject constructor(private val toastFactory: ToastFactory) : BasePresenter<AuthView>() {
 
-    fun onLoginButtonClicked() {}
+    fun onLoginButtonClicked(email: String, password: String) {
+        if (password.isEmpty()) view?.setPasswordError("This field is required")
+        if (email.isEmpty()) {
+            view?.setEmailError("This field is required")
```

Comment:
One way to avoid having strings around the code, which can be messy to modify, you can pass the resource of the string, like `R.strings.required_field_error`, and inside the method `setEmailError` call `context.getString(resourceId)`